include/zephyr/drivers/sensor.h

```diff
@@ -511,13 +528,38 @@ struct sensor_decoder_api {
 typedef int (*sensor_get_decoder_t)(const struct device *dev,
 				    const struct sensor_decoder_api **api);
 
+/**
+ * @brief Options for what to do with the associated data when a trigger is consumed
+ */
+enum sensor_stream_data_opt {
+	/** @brief Include whatever data is associated with the trigger */
+	SENSOR_STREAM_DATA_INCLUDE = 0,
+	/** @brief Do nothing with the associated trigger data, it may be consumed later */
+	SENSOR_STREAM_DATA_NOP = 1,
+	/** @brief Flush/clear whatever data is associated with the trigger */
+	SENSOR_STREAM_DATA_DROP = 2,
+};
+
+struct sensor_stream_trigger {
+	enum sensor_trigger_type trigger;
+	enum sensor_stream_data_opt opt;
+};
+
+#define SENSOR_STREAM_TRIGGER_PREP(_trigger, _opt)                                                 \
+	{                                                                                          \
+		.trigger = (_trigger), .opt = (_opt),                                              \
+	}
 /*
  * Internal data structure used to store information about the IODevice for async reading and
  * streaming sensor data.
  */
 struct sensor_read_config {
 	const struct device *sensor;
-	enum sensor_channel *const channels;
+	const bool is_streaming;
+	union {
```

Comment:
I don't think this should be a union, the reader may wish to select channels and trigger sources independently, e.g. on data ready read the gyro x and y channels only

Reply:
I wanted to avoid adding things that would/could get confusing. For example, lets say my application wants to know about the `RESET_DONE_INT`. I would have to add it to the triggers in sensor.h, then I could set the stream to tell me about it. I wouldn't set any channels. Does that mean I want to turn off the FIFO?

The current design is such that the read configs are minimally intrusive. Setting filters like that which affect the configuration registers should be done via attributes. In your example specifically:
* If the application only cares about reading the gyro data they can set the accel ODR to 0Hz
* If the application wants to batch gyro and poll accel they can set the ODR for both and add a PR to add new attributes to configure the FIFO to exclude channels.